### PR TITLE
Fix low level functions

### LINKED ISSUE #377: Method get_symbols() does not return all symbols
I'm using the python binding of this library. The method `get_symbols()` in `libmata.nfa.nfa` returns (as it seems) only up to 64 symbols. If the alphabet contains more symbols, they are omitted. However, the method `get_alphabet_symbols()` in `libmata.alphabets.OnTheFlyAlphabet` seems to work fine.

Example of an offending automaton:
https://github.com/VeriFIT/nfa-bench/blob/master/benchmarks/z3-noodler/automatark/intersection/instance06314-5.mata

Code generating this issue:
```
from libmata import parser, alphabets

alpha = alphabets.OnTheFlyAlphabet()
aut = parser.from_mata("/path/to/automaton.mata", alpha)

print(alpha.get_alphabet_symbols()) # returns symbols 0-77 for the automaton above
print(aut.get_symbols()) # returns symbols 0-63
```

I'm using version 1.2.0 of the python binding, installed through pip install. My OS is Ubuntu 22.04.3 LTS.

### LINKED ISSUE #378: Error in `get_symbol_map()`, python binding
An error occurs whenever running `get_symbol_map()` on a non-empty alphabet.

When running the following code:
```
from libmata import alphabets
alpha = alphabets.OnTheFlyAlphabet.from_symbol_map({'a': 0})
alpha.get_symbol_map()
```
this error occurs:
```
File libmata/alphabets.pyx:85, in libmata.alphabets.OnTheFlyAlphabet.get_symbol_map()

AttributeError: 'bytes' object has no attribute 'encode'
```
I'm using version 1.2.0 of the python binding, installed through pip install. My OS is Ubuntu 22.04.3 LTS.

### PR DESCRIPTION
**Pull request by Adda0**
This PR fixes some low-level functions which produced bugs, discovered in Python binding in #377 and #378.

I had to revert to `get_used_states()` using the slightly slower implementation with normal vector, as using bit vector will fail when using epsilons: it will try to allocate space for epsilon number of symbols. Optimizing the `get_used_states()` will need to be resolved again later.

This PR therefore fixes #377, and fixes #378.

### CODE REVIEW

File: src/nfa/delta.cc [hunk 1]
```diff
@@ -547,12 +547,12 @@ OrdVector<Symbol> Delta::get_used_symbols() const {
     //WITH BOOL VECTOR (error !!!!!!!):
     //return utils::OrdVector<Symbol>(utils::NumberPredicate<Symbol>(get_used_symbols_bv()));
 
-    //WITH BOOL VECTOR (1.9s):
-    std::vector<bool> bv{ get_used_symbols_bv() };
-    utils::OrdVector<Symbol> ov{};
-    const size_t bv_size{ bv.size() };
-    for (Symbol i{ 0 }; i < bv_size; ++i) { if (bv[i]) { ov.push_back(i); } }
-    return ov;
+    //WITH BOOL VECTOR (1.9s): (The fastest, it seems.)
```
Comment (jurajsic) [inline src/nfa/delta.cc:550]: I would add some note for the reason it fails (epsilon is too large).

**Review by jurajsic — APPROVED** [review]
Looks good. It should probably be easy to add some check that if there is no large symbol, we can use bool vectors.

File: src/nfa/delta.cc (hunk 1 above)
Comment (Adda0) [reply src/nfa/delta.cc:550]: Good idea. I added a comment.

**Comment by Adda0** [comment]
> It should probably be easy to add some check that if there is no large symbol, we can use bool vectors.

I made a note which I will convert into an issue later.

**Pull request merged**